osu.Game/Screens/OnlinePlay/Playlists/PlaylistsResultsScreen.cs

```diff
@@ -77,7 +80,50 @@ private void load()
             });
         }
 
-        protected override APIRequest FetchScores(Action<IEnumerable<ScoreInfo>> scoresCallback)
+        protected override void PopulateScorePanelList()
+        {
+            var req = FetchScores(fetchScoresCallback);
+
+            if (req != null)
+                api.Queue(req);
+        }
+
+        protected override void Update()
+        {
+            base.Update();
+
+            if (lastFetchCompleted)
+            {
+                APIRequest nextPageRequest = null;
+
+                if (ScorePanelList.IsScrolledToStart)
+                    nextPageRequest = FetchNextPage(-1, fetchScoresCallback);
+                else if (ScorePanelList.IsScrolledToEnd)
+                    nextPageRequest = FetchNextPage(1, fetchScoresCallback);
+
+                if (nextPageRequest != null)
+                {
+                    lastFetchCompleted = false;
+                    api.Queue(nextPageRequest);
+                }
+            }
+        }
```

Comment:
What is happening here? Why is this in update? Why is it not being handled like every single score request flow in every other results screen implementation?

Reply:
I moved this from the previous implementation of `ResultsScreen`, it allowed fetching next pages of scores. However, it only really made sense for `PlaylistResultsScreen`, for other implementations of `FetchNextPage` only returned null. Thus, I moved it.

Alternatively, I can make `MultiPageResultsScreen`, which implement this.